automatic/python3/update.ps1

```diff
@@ -12,52 +14,146 @@ function global:au_SearchReplace {
+  rm tools\*.msi, tools\*.exe -ea 0
 
-function GetStreams() {
-  param($releaseUrls)
-  $streams = @{ }
+  Get-RemoteFiles -Purge -NoSuffix
 
-  $releaseUrls | % {
-    $version = $_.InnerText.Trim() -split ' ' | select -Index 1
-    #if ($version -match '[a-z]') { Write-Host "Skipping prerelease: '$version'"; return }
-    $versionTwoPart = $version -replace '([\d]+\.[\d]+).*', "`$1"
+  # download Python documentation archive
+  $webrequest = [System.Net.HttpWebRequest]::Create($Latest.ZipUrl)
+  $response_stream = $webrequest.GetResponse().GetResponseStream()
+  $zip = [IO.Compression.ZipArchive]::new($response_stream)
 
-    if ($streams.$versionTwoPart) { return }
+  # license text for legal/LICENSE.txt
+  $license_entry = $zip.GetEntry("$($Latest.ZipName)/license.txt")
+  $Latest.License = [System.IO.StreamReader]::new($license_entry.Open()).ReadToEnd()
 
-    $download_page = Invoke-WebRequest -Uri "https://www.python.org$($_.href)" -UseBasicParsing
+  # copyright information for nuspec
+  $copyright_entry = $zip.GetEntry("$($Latest.ZipName)/copyright.txt")
+  $reader = [System.IO.StreamReader]::new($copyright_entry.Open())
+  (1..5) | % {$reader.ReadLine()}  # skip header
+  $copyright = ''
+  $reading_copyright = $false
+  while (($line = $reader.ReadLine()) -ne $null) {
+      if (!$line) {
+          $copyright += "`n"
+          $reading_copyright = $false
+          continue
+      }
+      if ($line.StartsWith('Copyright')) {
+          if ($reading_copyright) {
+              $copyright += "`n"
+          }
+          $copyright += $line
+          $reading_copyright = $true
+      } elseif ($reading_copyright) {
+          $copyright += " $line"
+      } else {
+          break
+      }
+  }
+  $Latest.Copyright = $copyright.Trim()
+}
+
+function global:au_AfterUpdate($Package) {
+    $cdata = $Package.NuspecXml.CreateCDataSection($Latest.Copyright)
+    $xml_Copyright = $Package.NuspecXml.GetElementsByTagName('copyright')[0]
+    $xml_Copyright.RemoveAll()
+    $xml_Copyright.AppendChild($cdata) | Out-Null
+    $Package.NuspecXml.package.metadata.licenseUrl = $Latest.LicenseUrl
+    $Package.NuspecXml.package.metadata.title = $Latest.Title
+    $Package.SaveNuspec()
+    $Latest.License | Out-File './legal/LICENSE.txt'
+}
+
+function GetStreams() {
+  param($release_files)
 
-    $url32 = $download_page.links | ? { $_.href -match "python-3.+.(exe|msi)$" -and $_.href -notmatch "amd64" } | select -first 1 -expand href
-    $url64 = $download_page.links | ? href -match "python-3.+amd64\.(exe|msi)$" | select -first 1 -expand href
-    if (!$url32 -or !$url64) {
-      Write-Host "Skipping due to missing installer: '$version'"; return
+  $version_re = '3\.(?<minor>\d+)\.(?<micro>\d+)(?:(?<releaselevel>a|b|rc)(?<serial>\d+))?'
+  $re = '^python-(?<version>' + $version_re + ')(?<amd64>-amd64)?\.exe$'
+  # collect URL pairs for all Python versions
+  $all_versions = @{ }
+  foreach ($file in $release_files) {
```

Comment:
We rarely use `foreach` loops in this repository and prefer piped calls.

If possible, change the foreach loop to `$release_files | ForEach-Object { }` instead.